rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -3020,10 +3046,79 @@ extern "C" rmw_ret_t rmw_wait(
 ///////////                                                                   ///////////
 /////////////////////////////////////////////////////////////////////////////////////////
 
+static rmw_ret_t get_matched_endpoints(
+  dds_entity_t h, dds_return_t (* fn)(
+    dds_entity_t h,
+    dds_instance_handle_t * xs, size_t nxs), std::vector<dds_instance_handle_t> & res)
```

Comment:
nit: defining the pointer to function before highly increases readability:

```cpp
using get_matched_endpoints_fn_t = dds_return_t (* fn)(
    dds_entity_t h,
    dds_instance_handle_t * xs, size_t nxs);
static rmw_ret_t get_matched_endpoints(
  dds_entity_t h, get_matched_endpoints_fn_t fn, std::vector<dds_instance_handle_t> & res)
```

Reply:
I must have seen too many C function pointer types ... :) but yes, that would be wise